test/unit/test_classical_explainer.py

```diff
@@ -15,21 +14,34 @@ class TestClassicalExplainer(object):
     def test_working(self):
         assert True
 
-    def test_explain_model_local(self):
+    def test_explain_model_local_default(self):
         """
         Test for explain_local of classical explainer
         :return:
         """
-        train_df = get_mnli_test_dataset('train')
-        X_str = train_df['sentence1']
-        ylabels = train_df['genre']
-        X_train, X_test, y_train, y_test = train_test_split(X_str, ylabels, train_size=0.8, test_size=0.2)
-
+        X_train, X_test, y_train, y_test = setup_mnli_test_train_split()
         label_encoder = LabelEncoder()
         y_train = label_encoder.fit_transform(y_train)
         explainer = ClassicalTextExplainer()
         classifier, best_params = explainer.fit(X_train, y_train)
         explainer.preprocessor.labelEncoder = label_encoder
 
-        local_explanantion = explainer.explain_local(DOCUMENT)
-        assert len(local_explanantion.local_importance_values) == len(local_explanantion.features)
+        local_explanation = explainer.explain_local(DOCUMENT)
+        assert len(local_explanation.local_importance_values) == len(local_explanation.features)
+
+    def test_explain_model_local_with_predicted_label(self):
```

Comment:
If Ilya takes a look at this file he'll suggest factoring more of the common pieces here. Personally, duped stuff over 2 tests doesn't bother me, but if these cases get expanded further it'll probably make sense to do some more refactoring

Reply:
Yeah - I also debated if I wanted to pull out the common piece but in future if we add more tests with the same code then we should separate it out.